cli/app.go

```diff
@@ -425,21 +426,21 @@ func (c *App) maybeRepositoryAction(act func(ctx context.Context, rep repo.Repos
 			defer gather.DumpStats(ctx)
 
 			if c.metricsListenAddr != "" {
-				mux := http.NewServeMux()
-				if err := initPrometheus(mux); err != nil {
+				m := mux.NewRouter()
+				if err := initPrometheus(m); err != nil {
 					return errors.Wrap(err, "unable to initialize prometheus.")
 				}
 
 				if c.enablePProf {
-					mux.HandleFunc("/debug/pprof/", pprof.Index)
-					mux.HandleFunc("/debug/pprof/cmdline", pprof.Cmdline)
-					mux.HandleFunc("/debug/pprof/profile", pprof.Profile)
-					mux.HandleFunc("/debug/pprof/symbol", pprof.Symbol)
-					mux.HandleFunc("/debug/pprof/trace", pprof.Trace)
+					m.HandleFunc("/debug/pprof/", pprof.Index)
```

Comment:
nit; might be slightly cleaner to create sub-router for pprof:

```go
pr := m.PathPrefix("/debug/pprof").
	Methods("POST", "GET").
	Schemes("http", "https").
	Subrouter()
pr.HandleFunc("/", pprof.Index)
pr.HandleFunc("/cmdline", pprof.Cmdline)
pr.HandleFunc("/profile", pprof.Profile)
pr.HandleFunc("/symbol", pprof.Symbol)
pr.HandleFunc("/trace", pprof.Trace)
```

Reply:
in my opinion it's easier to code-search for those strings if they are explicit and not mangled. For the same reason I use "/api/v1" prefix explicitly everywhere.

Reply:
fair enough